x/protocolpool/keeper/abci.go

```diff
@@ -0,0 +1,37 @@
+package keeper
+
+import (
+	"context"
+	"time"
+
+	"cosmossdk.io/x/protocolpool/types"
+
+	"github.com/cosmos/cosmos-sdk/telemetry"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func (k *Keeper) EndBlocker(ctx context.Context) error {
+	defer telemetry.ModuleMeasureSince(types.ModuleName, time.Now(), telemetry.MetricKeyEndBlocker)
+
+	logger := k.Logger(ctx).With("module", "x/"+types.ModuleName)
+
+	// Iterate over all continuous fund proposals and perform continuous distribution
+	err := k.ContinuousFund.Walk(ctx, nil, func(key sdk.AccAddress, value types.ContinuousFund) (bool, error) {
```

Comment:
Let's maybe add a comment, that it purposely loops through all payment streams and that too many payment streams can slow down blocks, but because it is gov gated that isn't a dos factor. This shows we are aware of it (for audits :D ).